packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -465,7 +463,9 @@ export abstract class BaseService extends Resource
     if (props.deploymentController?.type === DeploymentControllerType.EXTERNAL) {
       Annotations.of(this).addWarning('taskDefinition and launchType are blanked out when using external deployment controller.');
     }
-
+    if (props.circuitBreaker && props.deploymentController?.type !== DeploymentControllerType.ECS) {
+      Annotations.of(this).addWarning('Deployment circuit breaker requires the ECS deployment controller.');
```

Comment:
This should be `addError` if it is not allowed.

Reply:
Done